projecttacoma/deqm-test-server
# Added Winston Debug and Info Logging

# Summary
Added debug and info level logging through Winston to clarify the order or operations and assist with debugging.

## New behavior
- Running with NODE_ENV set to anything but 'debug' (defaults to 'production') will print only info-level logs
- Running with NODE_ENV set to 'debug' will print debug-level logs
- ndjsonWorker and importWorker now log using the existing logger instead of `console.log()`
- Logs will no longer print during unit tests

## Code changes
- Added a new .env variable to silence and trigger debug logging
- Added lots and lots of debug logging
- Added high-level info logging
- Added a jest config file to silence logging during unit tests

# Testing guidance
- Ensure all input and output to/from the server is unchanged
- Run using the command `NODE_ENV=development npm start` and ensure all debug-level logs appear
- Run using `npm start` and ensure only info-level logs appear
- Run `npm test` and ensure no logs appear
- Look through all files and determine if all potentially useful info is logged or if any unnecessary info is logged
